opentelemetry-sdk/tests/performance/benchmarks/metrics/test_benchmark_metrics.py

```diff
@@ -0,0 +1,77 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+
+from opentelemetry.sdk.metrics import Counter, MeterProvider
+from opentelemetry.sdk.metrics.export import (
+    AggregationTemporality,
+    InMemoryMetricReader,
+)
+
+reader = InMemoryMetricReader()
+reader_delta = InMemoryMetricReader(
+    preferred_temporality={
+        Counter: AggregationTemporality.DELTA,
+    },
+)
+provider = MeterProvider(
+    metric_readers=[reader],
+)
+provider2 = MeterProvider(metric_readers=[reader_delta])
+meter = provider.get_meter("sdk_meter_provider")
+meter2 = provider2.get_meter("sdk_meter_provider_delta")
+counter = meter.create_counter("test_counter")
+counter2 = meter2.create_counter("test_counter2")
+udcounter = meter.create_up_down_counter("test_udcounter")
```

Comment:
Should these variables be preserved across test runs or should they be injected (as a pytest fixture) for each time the test runs?

Reply:
I don't have a strong opinion. I pretty much followed what [dot net](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Metrics/HistogramBenchmarks.cs) is doing for their benchmarking. Also through my testing, using separate or the same instruments across tests don't have much impact on performance.

Reply:
IMO it would be cleaner if the relevant setup was part of the test case and to avoid leaking anything between test runs. Not a blocking comment

Reply:
Well, the only part that gets benchmarked is whatever is inside `benchmark`, right? If that is the case, then creating new instruments or not should not impact performance :thinking: